extensions/typescript-language-features/src/features/referencesCodeLens.ts

```diff
@@ -75,6 +74,7 @@ class TypeScriptReferencesCodeLensProvider extends TypeScriptBaseCodeLensProvide
 				}
 			// fallthrough
 
+			case PConst.Kind.function:
```

Comment:
This needs to be controlled enabled by a new setting. Take a look at the `typescript.referencesCodeLens.enabled` setting. 

Also, I'm concerned that this may cause the code lens to show up in too many locations. Can you explain how you've tested this?

Reply:
Thank you for your feedback.

>This needs to be controlled enabled by a new setting. Take a look at the typescript.referencesCodeLens.enabled setting.

OK, I'll do this!

Just to make sure, you mean that I should create the setting like `typescript.referencesInnerFunctionCodeLens.enabled`?

Reply:
>Also, I'm concerned that this may cause the code lens to show up in too many locations. Can you explain how you've tested this?

I tested this with TypeScript code (eg. https://github.com/microsoft/TypeScript/blob/master/src/compiler/checker.ts , https://github.com/microsoft/TypeScript/blob/master/src/compiler/factory.ts ).

For me, it does not seem too much nor cause performance bottlenecks.
(Because it does not show references for functions defined by `const`/`let`, but only for `function func()`?)

But if you have any ideas for filtering, please let me know.

Reply:
Yes, but call the new settings `typescript.referencesCodeLens.showOnAllFunctions`. (you'll also need to create `javascript.referencesCodeLens.showOnAllFunctions` setting)

Testing this against the TS repo is a good start. We can merge this once the setting is added and see if user have any feedback on the code lens showing up too often

Reply:
@mjbvz 
Hi, I added a new setting, please take a look at it?